google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java

```diff
@@ -0,0 +1,263 @@
+   * setup starting date time is set to one day after the last. If not, the billing setup is set to
+   * start immediately. The ending date is set to one day after the starting date time.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID containing the BillingSetup to remove.
+   * @param billingSetup The instance of BillingSetup whose starting date time will be set.
+   * @return Billing Setup with updated start and end date time.
+   */
+  private BillingSetup setBillingSetupDateTimes(
+      GoogleAdsClient googleAdsClient, long customerId, BillingSetup billingSetup)
+      throws Exception {
+    // The query to search existing approved billing setups in the end date time descending
+    // order.
+    // See GetBillingSetup.cs for a more detailed example of requesting billing setup
+    // information.
+    String query =
+        "SELECT billing_setup.end_date_time "
+            + "FROM billing_setup "
+            + "WHERE billing_setup.status = 'APPROVED' "
+            + "ORDER BY billing_setup.end_date_time DESC";
+
+    // Creates the service client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues a search request.
+      SearchPagedResponse searchPagedResponse =
+          googleAdsServiceClient.search(Long.toString(customerId), query);
+
+      BillingSetup.Builder billingSetupBuilder = billingSetup.toBuilder();
+      if (searchPagedResponse.getPage().getResponse().getResultsCount() > 0) {
+        // Retrieves the ending date time of the last billing setup.
+        StringValue lastEndingDateTimeString =
+            searchPagedResponse
+                .getPage()
+                .getResponse()
+                .getResults(0)
+                .getBillingSetup()
+                .getEndDateTime();
+
+        // A null ending date time indicates that the current billing setup is set to run
+        // indefinitely. Billing setups cannot overlap, so throw an exception in this case.
+        if (lastEndingDateTimeString == null) {
+          throw new Exception(
+              "Cannot set starting and ending date times for the new billing setup; the latest"
+                  + " existing billing setup is set to run indefinitely.");
+        }
+
+        DateTime lastEndingDateTime = DateTime.parse(lastEndingDateTimeString.toString());
+
+        // Sets the new billing setup to start one day after the ending date time.
+        billingSetupBuilder.setStartDateTime(
+            StringValue.of(lastEndingDateTime.plusDays(1).toString("yyyy-MM-dd")));
+
+        // Sets the new billing setup to end one day after the starting date time.
+        billingSetupBuilder.setEndDateTime(
+            StringValue.of(lastEndingDateTime.plusDays(2).toString("yyyy-MM-dd")));
+      } else {
+        // Otherwise, the only acceptable start time is TimeType.Now.
```

Comment:
For clarity, perhaps "the only acceptable start time is DateTime.now()"?  TimeType.Now is the C# value.